### ISSUE TITLE
imap relies on a C locale for date formatting

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3065]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)| @therve reported|
|-|-|
|Trac ID|trac#3065|
|Type|defect|
|Created|2008-03-03 17:35:05Z|

spew_internaldate uses time.strftime, which is locale dependant. It should use something not relying on the locale.

The problem can be reproduce by running the imap test suite with gtk2 reactor and a non english locale (like fr_FR).

<details><summary>Searchable metadata</summary>

```
trac-id__3065 3065
type__defect defect
reporter__therve therve
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__duplicate duplicate
component__mail mail
keywords__locale locale
time__1204565705000000 1204565705000000
changetime__1344869190000000 1344869190000000
version__None None
owner__argonemyth argonemyth
cc__moijes12
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

This is actually a bug in the implementation of date formatting, not in the tests.  The resulting string _should not_ vary depending on the locale of the server environment.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve commented|
|-|-|

Yeah it seems so. I've looked at the RFC (http://www.faqs.org/rfcs/rfc3501.html), and the format date-time uses date-month, and date-month explicitly lists the month in english abbreviated format.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="moijes12's avatar" src="https://avatars.githubusercontent.com/u/4566851?s=50" width="50" height="50">](https://github.com/moijes12)<a name="note_6"></a>|@moijes12 commented|
|-|-|

So, if someone were to start working on this, where does he begin ?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun commented|
|-|-|

Thanks for your interest in this ticket, moijes12.  Can you explain what you're asking in more detail?  Superficially, it seems like this ticket describes the issue in pretty good detail.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @argonemyth|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="argonemyth's avatar" src="https://avatars.githubusercontent.com/u/965739?s=50" width="50" height="50">](https://github.com/argonemyth)<a name="note_9"></a>|@argonemyth set status to `closed`|
|-|-|

From what I read in the method `spew_internaldate`, someone already fixed the problem and we already have a test for it: `test_fetchInternalDateLocaleIndependent`.

Here is my local output from trial [trunk r35167]:

```
$ bin/trial -r gtk2 twisted.mail.test.test_imap.NewFetchTestCase.test_fetchInternalDateLocaleIndependent
twisted.mail.test.test_imap
  NewFetchTestCase
    test_fetchInternalDateLocaleIndependent ...                            [OK]

-------------------------------------------------------------------------------
Ran 1 tests in 0.059s

PASSED (successes=1)
```

Just to make sure, I changed my locale to `fr_FR.utf8` and ran the mail test suite with gtk2 reactor, all tests passed. I am going to close this ticket, if anyone still run into this issue, please re-open the ticket.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun set status to `reopened`|
|-|-|

This was fixed by [[#4937](https://github.com/twisted/twisted/issues/4937)](https://github.com/twisted/twisted/issues/4937), which was a duplicate of this ticket.  Marking this one as the duplicate, since #4937 is already resolved.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set status to `closed`|
|-|-|
